spec.html

```diff
@@ -686,23 +812,40 @@ <h1><ins>AsyncModuleExecutionFulfilled ( _module_ )</ins></h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-AsyncModuleExecutionRejected" aoid="AsyncModuleExecutionRejected">
+    <emu-clause id="sec-gathercyclemodules" aoid="GatherCycleModules">
+      <h1><ins>GatherCycleModules ( _module_, _ancestorIndex_, _cycleModules_ )</ins></h1>
+      <emu-alg>
+        1. Assert: _module_.[[Status]] is ~evaluated~.
+        1. Assert: _module_.[[DFSAncestorIndex]] is _ancestorIndex_.
+        1. For each String _required_ that is an element of _module_.[[RequestedModules]], do
+          1. Let _requiredModule_ be ! HostResolveImportedModule(_module_, _required_).
+          1. If _requiredModule_.[[DFSAncestorIndex]] is equal to _ancestorIndex_ and _cycleModules_ does not contain _requiredModule_, then
+            1. Append _requiredModule_ to _cycleModules_.
+            1. Perform ! GatherCycleModules(_requiredModule_, _ancestorIndex_, _cycleModules_).
+        1. Return *undefined*.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+    <emu-clause id="sec-asyncmoduleexecutionrejected" aoid="AsyncModuleExecutionRejected">
       <h1><ins>AsyncModuleExecutionRejected ( _module_, _error_ )</ins></h1>
       <emu-alg>
         1. Assert: _module_.[[Status]] is ~evaluated~.
         1. If _module_.[[AsyncEvaluating]] is *false*,
           1. Assert: _module_.[[EvaluationError]] is not *undefined*.
           1. Return *undefined*.
         1. Assert: _module_.[[EvaluationError]] is *undefined*.
-        1. Set _module_.[[EvaluationError]] to ThrowCompletion(_error_).
-        1. Set _module_.[[AsyncEvaluating]] to *false*.
+        1. Let _cycleRoot_ be ! GetAsyncCycleRoot(_m_).
+        1. Assert: _cycleRoot_.[[EvaluationError]] is *undefined*.
+        1. Let _cycleModules_ be a new empty List.
+        1. Perform ! GatherCycleModules(_cycleRoot_, _cycleRoot_.[[DFSIndex]], _cycleModules_).
```

Comment:
It looks like this does solve the, I guess we can say fragility? That I came across in my issue. I want to work with this a bit more and make sure that I understand it. Will do that tomorrow.

Reply:
The root cause of the fragility here was most likely due to https://github.com/tc39/proposal-top-level-await/pull/161.